composer/utils/checkpoint.py

```diff
@@ -1067,13 +1064,12 @@ def _save_checkpoint(
         # Only rank 0 saves RNG
         if dist.get_global_rank() > 0:
             state_dict.pop('rng')
-        # To load optimizer states with 2.0 <= torch < 2.2.9 , the optimizer state must be at the top
-        # level of the state dict because the load_sharded_optimizer_state_dict function
+        # To load optimizer states with current torch versions, the optimizer state must be at the
+        # top level of the state dict because the load_sharded_optimizer_state_dict function
         # requires a top level state dict key for the optimizer.
         # See https://github.com/pytorch/pytorch/blob/v2.0.1/torch/distributed/checkpoint/optimizer.py#L271
         # for more info.
-        if version.parse(torch.__version__) < version.parse('2.2.9'):
-            state_dict['optimizers'] = state_dict['state'].pop('optimizers')
+        state_dict['optimizers'] = state_dict['state'].pop('optimizers')
```

Comment:
Do we need this change where we change state dict? I really dont like optimizers at root level.

Reply:
wait what those comments are from 2 years ago

Reply:
what makes you think they still hold?

Reply:
Can we try it without?

Reply:
this bug makes me think it's still valid at least lol. I'll test